### DISCUSSION TITLE
Can I create a dispatch in the slice instead of the view?

### DISCUSSION THREAD

**Discussion by DanPresa** [reduxjs/redux-toolkit#3325] [Q&A]
Hello everyone.
I have a question. Is it possible to create the dispatch in the slice (reducer) instead of the view?
A tech lead told me that once but. I don't know how to do it.

**Comment by markerikson** [comment]
What do you mean by "create a dispatch"?  Can you give an example of what you're trying to do?

**Reply by DanPresa** [reply]
Hello. I just found out an article in another forum and it says that it is an anti-pattern.
I meant if it was possible to make a dispatch in a reducer.

**Reply by markerikson** [reply]
No, it's absolutely _impossible_ to dispatch from a reducer. You're not allowed to do that, and Redux will throw an error if you do.

A reducer has one job: look at the current `state` and the `action`, and decide what the new `state` should be and return it.  That's all.
